Storage/Storage/Bookmarks.swift

```diff
@@ -34,14 +36,11 @@ public struct BookmarkRoots {
 /**
  * The immutable base interface for bookmarks and folders.
  */
-@objc public protocol BookmarkNode {
-    var guid: String { get }
-    var title: String { get }
-    var icon: UIImage { get }
-}
-
-@objc public protocol Bookmark : BookmarkNode {
-    var url: String! { get }
+public class BookmarkNode {
+    public var id: Int? = nil
```

Comment:
I've been doing this intentionally. If you want to do add a History entry or something, you create one with a url and title, but the db is responsible for giving it an ID. It will look one up if it needs it (and assign it to you for future reference).